### ISSUE TITLE
Android performance problems: slow, high power usage

### ISSUE THREAD

**Issue by nolanl** [zerotier/ZeroTierOne#535]
I've had Zerotier running on a Nexus 5x for about 6 hours since my last charge.  The ZT network is completely idle:

bullhead:/ $ ifconfig tun0
tun0 Link encap:UNSPEC
inet addr:192.168.192.147 P-t-P:192.168.192.147 Mask:255.255.255.0
UP POINTOPOINT RUNNING MTU:2800 Metric:1
RX packets:0 errors:0 dropped:0 overruns:0 frame:0
TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
collisions:0 txqueuelen:500
RX bytes:0 TX bytes:0
bullhead:/ $ 

But according to the battery settings page, Zerotier has used more power than everything except for the screen backlight!

I looked into this because my battery has stopped lasting through the entire day ever since I started running ZeroTier, even though I haven't really started putting real traffic through it yet...

**Comment by LFlare** [comment]
Experiencing the same problem. Accidentally left it on overnight and completely drained my battery.

**Comment by LFlare** [comment]
@adamierymenko, I reckon this is a pretty big problem. Would love to be able to keep my VPN on without battery drain. Is the Android application FOSS? I can take a look at it if it is...

**Comment by adamierymenko** [comment]
We're also investigating some performance issues on Android, and these may be related. Will keep you posted here.

**Comment by adamierymenko** [comment]
Closing since this is old and the next version has some major perf improvements on Android.

**Comment by M-Pixel** [comment]
Could this issue be re-opened?  This is still pretty bad.  I get about 8 hours battery with ZT on, over 48 hours without.  Phone gets hot and ZeroTier is shown as the majority power consumer.

**Comment by Vakrehus** [comment]
Still happens in 2021 on Pixel 4a

**Comment by ArcziHA** [comment]
Still happens in Samsung Smartphones (S9+, S21)
![Screenshot_20220104-160542_Device care](https://user-images.githubusercontent.com/72281532/148088334-92413f9d-173b-4f34-8f64-9a7d47eca4a2.jpg)
![IMG-20220104-WA0004](https://user-images.githubusercontent.com/72281532/148088343-f3706889-244f-4510-988c-cbe4ebc1dab6.jpg)

**Comment by mebitek** [comment]
Same on xiaomi redmi note 9 pro

**Comment by Rwin1234** [comment]
I love zerotier VPN, I use it to connect to my Home Assistant box at home, but battery drain is still very high. It drains 30-40% of my battery on two Samsung phones. Even if I dont allow it to run in the background. Once it is running e.g. when I connect it manually to check on my HA, it seems to keep running. I would expect it to go back to sleep when not in use.

**Comment by eregnier** [comment]
![Image](https://github.com/user-attachments/assets/e4819a24-d7f5-48e2-b0f8-4619fd3d8bf1)

Hey team, thank you for ZT, it's amazing.

This problem is not solved as of today. high battery drain prevents me to let ZT run in background, I always start app, toggle network on, do my ZT vpn network action, end session by disconnecting my device from network. 

I had the issue some time ago, and thought it was solved, so I tried again to let ZT connected in background.

There is almost no data over this network from my phone since last phone full battery charge. my phone ends up being hot.

What is the app doing wrong ? 

This is  the only pain point using of ZT in my opinion. I hope this can be fixed any soon.

My device is google pixel 6 pro, Android 15

Thank you for ZT and any work to solve this.
